ports/android/README.md

```diff
@@ -0,0 +1,149 @@
+        -DANDROID_ABI=x86 -DANDROID_NATIVE_API_LEVEL=28 -DBUILD_SHARED_LIBS=Off ..
+$ cmake --build .
+```
+
+> Android supports shared libraries. `BUILD_SHARED_LIBS` is disabled solely
+> for demonstration purposes.
+
+
+## Setting up a networking bridge
+
+Each emulator instance runs behind a virtual router/firewall service that
+isolates it from the host. The Android version running in the emulator can
+communicate with the host over the loopback interface using the specialized
+alias network address *10.0.2.2*, but IGMP and multicast are not supported.
+To communicate with an applications not running in the QEMU guest an
+additional network interface is required.
+
+ * Create the network bridge and a *tap* device for the emulator to use
+```
+$ ip link add name br0 type bridge
+$ ip link set br0 up
+$ ip addr add 192.168.178.1/24 dev br0
+$ ip tuntap add dev tap0 mode tap user $USER
+$ ip link set tap0 master br0
+$ ip link set dev tap0 up promisc on
+```
+
+ * Launch the emulator with the *tap* device attached
+```
+$ <path/to/Android/Sdk>/emulator/emulator -avd Pixel_XL_API_29 -netdelay none -netspeed full -qemu -device virtio-net-pci,netdev=eth1 -netdev tap,id=eth1,ifname=tap0,script=no
+```
+
+ * Open a shell in the emulator
+```
+$ <path/to/Android/Sdk>/platform-tools/adb shell
+```
+
+ * Execute `su` from the shell to become *root* and configure the interface.
+ * Bring up the interface and assign an IP address.
+```
+# ip link set dev eth1 up
+# ip addr add 192.168.178.2/24 dev eth1
+```
+
+ * Make normal routing take precedence over anything that google/qemu configured.
+```
+$ ip rule add from all lookup main pref 99
+```
+
+ * You should now be able to ping the host from the emulator and vice versa.
+
+> The [KDE Community Wiki](5) is an excellent source of information on the
+> Android emulator. The second-to-last instruction stems from there and was
+> vital to establish communication between the emulator and the host.
+
+[5]: https://community.kde.org/KDEConnect/Android_Emulator
+
+
+## Deploying and running the RoundTrip example
+
+ * Push the `RoundTripPong` binary and open a shell.
+```
+$ <path/to/Android/Sdk>/platform-tools push <path/to/cyclonedds>/build.android/bin/RoundTripPong /data/local/tmp
+$ <path/to/Android/Sdk>/platform-tools shell
+```
+
+> The binary must be copied to the local filesystem, not the sd-card, as that
+> is mounted with the `noexec` flag and will not allow you to execute any
+> executables.
+
+ * Make the binary executable (from the shell on the device).
+```
+$ cd /data/local/tmp
+$ chmod 755 RoundTripPong
+```
+
+ * Force the guest to use the `eth0` interface.
+```
+$ export CYCLONEDDS_URI='<CycloneDDS><Domain><General><NetworkInterfaceAddress>eth0</NetworkInterfaceAddress></General></Domain></CycloneDDS>'
+```
+
+ * Execute `$ ./RoundTripPong`
+ * Build Eclipse Cyclone DDS for your host machine
```

Comment:
remove the sections on building for host & setting up networking OR move them to the end as fyi